src/components/DuplicateAgreementModal/DuplicateAgreementModal.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isEmpty, mapValues } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import {
+  Button,
+  Checkbox,
+  Modal,
+  ModalFooter
+} from '@folio/stripes/components';
+
+export default class DuplicateAgreementModal extends React.Component {
+  static propTypes = {
+    onClose: PropTypes.func,
+    onClone: PropTypes.func,
+  };
+
+  constructor() {
+    super();
+
+    this.cloneableProperties = {
+      'agreementInfo': <FormattedMessage id="ui-agreements.duplicateAgreementModal.agreementInfo" />,
+      'internalContacts': <FormattedMessage id="ui-agreements.duplicateAgreementModal.internalContacts" />,
+      'agreementLines': <FormattedMessage id="ui-agreements.duplicateAgreementModal.agreementLines" />,
+      'linkedLicenses': <FormattedMessage id="ui-agreements.duplicateAgreementModal.linkedLicenses" />,
+      'externalLicenses': <FormattedMessage id="ui-agreements.duplicateAgreementModal.externalLicenses" />,
+      'organizations': <FormattedMessage id="ui-agreements.duplicateAgreementModal.organizations" />,
+      'supplementaryInformation': <FormattedMessage id="ui-agreements.duplicateAgreementModal.supplementaryInformation" />,
+      'usageData': <FormattedMessage id="ui-agreements.duplicateAgreementModal.usageData" />
+    };
+
+    this.state = {
+      allSelected: false,
+      selected: {},
+    };
+  }
+
+  updateSelection = (e) => {
+    const name = e.target.name;
+    const isChecked = e.target.checked;
+    this.setState(prevState => (
+      {
+        selected: { ...prevState.selected, [name]: isChecked },
+        allSelected: isChecked ? prevState.allSelected : false,
```

Comment:
Take the following scenario:
- Load the modal, nothing is checked.
- One by one, check every individual checkbox.
- At this point, one would assume the `allSelected` value and top-level checkbox would be selected. But it's not.

You can take a look at how [Basket and BasketList](https://github.com/folio-org/ui-erm/blob/1b0fea5d0202ab481823c0275691e92dc7b704ff/src/components/views/Basket.js#L209) solve this problem if you want. That is, the displayed value of the select-all checkbox is calculated during each render.